File: backend/app/schemas/analytics.py
"""Analytics response schemas."""

from pydantic import BaseModel


class MetricSummary(BaseModel):
    total: int
    active: int
    completed: int
    average_per_conversation: float


class LeadMetrics(BaseModel):
    total: int
    new: int
    qualified: int
    converted: int


class RankedQuestion(BaseModel):
    question: str
    count: int


class AnalyticsOverview(BaseModel):
    conversations: MetricSummary
    messages: MetricSummary
    leads: LeadMetrics
    conversion_rate: float
    popular_questions: list[RankedQuestion]
    unanswered_questions: list[RankedQuestion]

File: frontend/src/api/analytics.ts
import { get } from "./client";

export interface RankedQuestion {
  question: string;
  count: number;
}

export interface AnalyticsOverview {
  conversations: {
    total: number;
    active: number;
    completed: number;
    average_per_conversation: number;
  };
  messages: {
    total: number;
    active: number;
    completed: number;
    average_per_conversation: number;
  };
  leads: {
    total: number;
    new: number;
    qualified: number;
    converted: number;
  };
  conversion_rate: number;
  popular_questions: RankedQuestion[];
  unanswered_questions: RankedQuestion[];
}

export function getAnalyticsOverview(): Promise<AnalyticsOverview> {
  return get<AnalyticsOverview>("/analytics/overview");
}
